lab_6_pipeline/pipeline.py

```diff
@@ -189,6 +324,16 @@ def _count_frequencies(self, article: Article) -> dict[str, int]:
         Returns:
             dict[str, int]: POS frequencies
         """
+        pos_freq = {}
+
+        for conllu_sent in self._analyzer.from_conllu(article=article).sentences:
+            word_features = [word.to_dict().get('upos') for word in conllu_sent.words]
+
+            for word in set(word_features):
+                pos_freq[word] = pos_freq.get(word, 0) + word_features.count(word)
```

Comment:
question 1: is it so mandatory to make:

1. list
2. set
3. loop over set
4. run count for each set member?

Reply:
I create a list to be able to store information about each word. The set and the loop over the set are created to avoid going over the same word several times. The count method is not used for set items, but for list items.